src/index.ts

```diff
@@ -182,30 +224,63 @@ function main() {
 
                 const deployUI = DeployUI.instance;
                 let deploymentManager: IDeploymentManager;
-                return prompt(questions.value)
+                let subPrompt: Promise<Answers>;
+                if (program.subscriptionId) {
+                    subPrompt = Promise.resolve<Answers>({ subscriptionId: program.subscriptionId});
+                } else {
+                    subPrompt = prompt(questions.value);
+                }
+                return subPrompt
                 .then((ans: Answers) => {
                     answers = ans;
-                    const index = cachedAuthResponse.subscriptions.findIndex((x: LinkedSubscription) => x.id === answers.subscriptionId);
+                    const index = cachedAuthResponse.linkedSubscriptions.findIndex((x: LinkedSubscription) => x.id === answers.subscriptionId);
                     if (index === -1) {
                         const errorMessage = 'Selected subscriptionId was not found in cache';
                         console.log(errorMessage);
                         throw new Error(errorMessage);
                     }
-                    cachedAuthResponse.options.domain = cachedAuthResponse.subscriptions[index].tenantId;
-                    answers.domainName = cachedAuthResponse.options.domain;
-                    deploymentManager = new DeploymentManager(cachedAuthResponse.options, answers.subscriptionId, program.type, program.sku);
+                    cachedAuthResponse.credentials.domain = cachedAuthResponse.linkedSubscriptions[index].tenantId;
+                    answers.domainName = cachedAuthResponse.credentials.domain;
+                    const serviceClientCredentials: ServiceClientCredentials = cachedAuthResponse.isServicePrincipal ?
+                        cachedAuthResponse.credentials : new DeviceTokenCredentials(cachedAuthResponse.credentials);
+                    deploymentManager = new DeploymentManager(
+                        serviceClientCredentials,
+                        cachedAuthResponse.credentials.environment,
+                        answers.subscriptionId,
+                        program.type,
+                        program.sku);
                     return deploymentManager.getLocations();
                 })
                 .then((locations: string[] | undefined) => {
-                    if (locations && locations.length > 0) {
+                    if (program.location && (program.websiteName || program.solutionName) && program.username) {
```

Comment:
if one of program.websiteName or program.solutionName is null we should set them to the non-null value and use below for azureWebsiteName and solutionName

Reply:
@zhang-hua Can you add a todo to fix this later?